app/views/actions/_pin.html.haml

```diff
@@ -0,0 +1,14 @@
+- if answer.pinned?
+  = button_to unpin_answer_path(id: answer.id),
+    class: "dropdown-item",
+    method: :delete,
+    form: { id: "ab-pin-#{answer.id}", data: { turbo_stream: true } } do
+    %i.fa.fa-fw.fa-thumbtack
+    = t(".unpin")
+- else
+  = button_to pin_answer_path(id: answer.id),
```

Comment:
i get a

```
ActionController::UrlGenerationError - No route matches {:action=>"pin", :controller=>"answer", :id=>109842029517652183}, missing required keys: [:username]
Did you mean?  pin_answer_url
               unpin_answer_url
               unpin_answer_path:
  app/views/actions/_pin.html.haml:9
  app/views/actions/_answer.html.haml:20
  app/views/answerbox/_actions.html.haml:16
  app/views/application/_answerbox.html.haml:24
  app/views/timeline/timeline.html.haml:3
  app/views/timeline/timeline.html.haml:2
  app/controllers/timeline_controller.rb:39:in `block (2 levels) in paginate_timeline'
  app/controllers/timeline_controller.rb:38:in `paginate_timeline'
  app/controllers/timeline_controller.rb:9:in `index'
  app/controllers/application_controller.rb:23:in `switch_locale'
```

when trying to view my timeline and having a pinned answer

Reply:
c3e48b7